packages/babel-plugin-transform-parameters/src/rest.ts

```diff
@@ -243,6 +243,12 @@ export default function convertFunctionRest(path) {
   let rest = node.params.pop().argument;
 
   if (rest.name === "arguments") scope.rename(rest.name);
+  if (rest.type === "ArrayPattern") {
+    rest.elements.forEach(
+      element =>
+        (element.name = scope.generateUidIdentifier(element.name).name),
+    );
+  }
```

Comment:
The array pattern can have nested binding patterns, for example:
```
function foo(...[...{ length: x }]) {}
```
Traversing a binding pattern can be daunting, instead we can check the binding info. For each `param`-kind binding in the function scope, we can check if it has non-empty `.constantViolations`. If the violation path is not the function itself (in cases such as `(function x(...[x]) {})`), then we can rename the param.

Reply:
Okay, I got it I will check for the constantViolations forEach param bindings.